### PR TITLE
fix: polish secure view

### LINKED ISSUE #9392: [ocis][web] polish secure view
# Description

## User Stories

* > As a ocis dev i need to have the dev stack so that i can develop/debug secure view.
*  > As an ocis qa engingeer i need coninously deployment examples to confirm the expected features.

## Acceptance Criteria
- update wopi example
- update docker compose web dev setup web
- new wopi server needs to be configured (web-latest)

Note:
- drone integration is already done
- we consider the code to be ready, if not -> bug report
- docs: https://owncloud.dev/services/collaboration/

## Definition of ready
- [ ] Everybody needs to understand the value written in the user story
- [ ] Acceptance criteria have to be defined
- [ ] All dependencies of the user story need to be identified
- [ ] Feature should be seen from an end user perspective
- [ ] Story has to be estimated
- [ ] Story points need to be less than 20

## Definition of done
- Functional requirements
  - [ ] Functionality described in the user story works
  - [ ] Acceptance criteria are fulfilled
- Quality
  - [ ] Code review happened
  - [ ] CI is green (that includes new and existing automated tests)
  - [ ] Critical code received unit tests by the developer
- Non-functional requirements
  - [ ] No sonar cloud issues

### PR DESCRIPTION
**Pull request by micbar**
## Description

Polish Secure view

1. Fix `view_comment` mode for pdfs - needed for Secure Viewable PDF files
2. Remove the secure viewer role from space roots.

## Related Issue

- Fixes #9392 

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: services/collaboration/pkg/service/grpc/v0/service.go [hunk 1]
```diff
@@ -122,7 +128,12 @@ func (s *Service) OpenInApp(
 		// the URL of the end-user application in view mode when different (defaults to edit mod URL)
 		viewAppURL = editAppURL
 	}
-
+	if viewAppURL == "" && editAppURL == "" && viewCommentAppURL != "" {
+		// there are rare cases where neither view nor edit is supported but view_comment is
+		// that can be the case for editable and viewable files
+		editAppURL = viewCommentAppURL
+		viewAppURL = viewCommentAppURL
+	}
```
[the remark this answers is not in the document]
Comment (jvillafanez) [reply services/collaboration/pkg/service/grpc/v0/service.go:136]: I've checked that Collabora only have the "view_comment" action for pdfs, but OnlyOffice has both "view" and "edit" actions (no "view_comment"). Still fragile if Collabora decides to include an "edit" action for pdfs at some point.

[the remark this answers is not in the document]
Comment (micbar) [reply services/collaboration/pkg/service/grpc/v0/service.go:136]: > This feels weird. Even if we assume the `viewCommentAppURL` will be used for secure view, it's weird that it's used if no edit or view urls are provided. I mean, maybe we should use the view_comment url under some conditions despite of having a valid view url.
> 

I thought that too. But there are still "edit" and "view" modes. In edit mode, comments should be written to the file, in view mode, you can only view.

> It seems better to move it below, where the actual `appURL` is being assigned, so we can set it to the view_comment url if needed without overwriting additional variables. I still think we're missing some condition to use that url for secure view

I put it here becuause the condition needs to be checked if we have view or edit access. Like said, that also applies for editing.

[the remark this answers is not in the document]
Comment (micbar) [reply services/collaboration/pkg/service/grpc/v0/service.go]: I adapted it a bit and added a TODO

**Review by jvillafanez — APPROVED** [review]
let's go with this for now. I'll try to cleanup the service file at some point.

**Pull request merged**